### PR TITLE
Add OSPlatform entries for iOS/tvOS/watchOS/Android

### PR DESCRIPTION
**Pull request by akoeplinger**
Implements the new OSPlatform members from https://github.com/dotnet/runtime/issues/33331

### CODE REVIEW

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs [hunk 1]
```diff
@@ -19,10 +19,14 @@ public enum Architecture
...
         private readonly object _dummy;
         private readonly int _dummyPrimitive;
+        public static System.Runtime.InteropServices.OSPlatform Android { get { throw null; } }
         public static System.Runtime.InteropServices.OSPlatform Browser { get { throw null; } }
         public static System.Runtime.InteropServices.OSPlatform FreeBSD { get { throw null; } }
+        public static System.Runtime.InteropServices.OSPlatform iOS { get { throw null; } }
         public static System.Runtime.InteropServices.OSPlatform Linux { get { throw null; } }
         public static System.Runtime.InteropServices.OSPlatform OSX { get { throw null; } }
+        public static System.Runtime.InteropServices.OSPlatform tvOS { get { throw null; } }
+        public static System.Runtime.InteropServices.OSPlatform watchOS { get { throw null; } }
```
Comment (stephentoub) [inline src/libraries/System.Runtime.InteropServices.RuntimeInformation/ref/System.Runtime.InteropServices.RuntimeInformation.cs:29]: @terrajobst, are all of these approved and ready to go?

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs [hunk 2]
```diff
@@ -16,6 +18,12 @@ namespace System.Runtime.InteropServices
 
         public static OSPlatform OSX { get; } = new OSPlatform("OSX");
 
+        public static OSPlatform iOS { get; } = new OSPlatform("IOS");
+
+        public static OSPlatform tvOS { get; } = new OSPlatform("TVOS");
+
+        public static OSPlatform watchOS { get; } = new OSPlatform("WATCHOS");
```
Comment (stephentoub) [inline src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs:25]: Assuming these were approved, LGTM.

**Comment by akoeplinger** [comment]
The APIs were approved in https://github.com/dotnet/runtime/issues/33331.

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs (hunk 2 above)
Comment (stephentoub) [inline src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs:25]: Tests?

Comment (akoeplinger) [reply src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs:24]: @stephentoub added them.

**Review by stephentoub — APPROVED** [review]

**Pull request merged**